### ISSUE TITLE
v1 Training time + training a 44Khz model on Universal VCTK+Blizzard2011+Clipper Datasets

### ISSUE THREAD

**Issue by CookiePPP** [jik876/hifi-gan#7]
What's a rough estimate for training time of the 22Khz v1 pretrained model provided here?

---

I'm currently training a 44Khz config.
At the moment I'm training using 3 GPU's, `batch_size=24` and `segment_size=16384` for a total `1179648` samples per iter.
Not sure how long I should leave it running, any rough estimate would be awesome.

Thanks!

**Comment by jik876** [comment]
> What's a rough estimate for training time of the 22Khz v1 pretrained model provided here?
> 
> I'm currently training a 44Khz config.
> At the moment I'm training using 3 GPU's, `batch_size=24` and `segment_size=16384` for a total `1179648` samples per iter.
> Not sure how long I should leave it running, any rough estimate would be awesome.
> 
> Thanks!

It took about 13-14 days to train the model up to 2,500k steps with two V100 GPUs.
I think the settings you mentioned will be much slower than my experience.
It is advisable to adjust the training steps as needed because it synthesizes high-quality audio even at an earlier step.

**Comment by CookiePPP** [comment]
Thanks.
I'll leave this issue open for now and rename it.
(Hopefully) someone will be interested in higher sampling rates using this vocoder and I can provide some side-by-side audio samples and a pretrained model.

**Comment by Yablon** [comment]
@jik876 Hello, I think maybe 13-14 days is too long to change some settings and get some new results. 

Do you have any idea about how to accelerate the whole training process ? What about pretrained models by stft and mel loss on generator ?

Thank you !

**Comment by jik876** [comment]
@Yablon 

Thanks for your interest.
Transfer learning a previously trained model would be a good way.
In our experience, transfer learning has been well applied to HiFi-GAN, so I think you will get good results without having to train the model from scratch when you change settings. In some cases this can lead to lower quality than training from scratch, but it would be enough to make sure that the settings you have changed are effective.
Except for using more GPUs, I think you have to take a trade-off in quality to accelerate training.

**Comment by rishikksh20** [comment]
@CookiePPP What are the hyper-parameters you are using for training 44.1 kHz ?
Specially these:
```
"num_mels": 
 "num_freq": 
 "n_fft": 
 "hop_size":
 "win_size": 
```

**Comment by CookiePPP** [comment]
@rishikksh20 
https://github.com/CookiePPP/cookietts/blob/experimental/CookieTTS/_4_mtw/hifigan/config_v1_48Khz_multiGPU.json#L16-L29

```json
    "discriminator_periods": [3, 5, 7, 11, 17, 23, 37],
    
    "segment_size": 16384,
    "num_mels": 80,
    "num_freq": 1025,
    "n_fft"   : 2048,
    "hop_size": 512,
    "win_size": 2048,

    "sampling_rate": 44100,

    "fmin": 20,
    "fmax": 11025,
    "fmax_for_loss": null,
```

**Comment by CookiePPP** [comment]
Closing this, comment here if you want any updates or models as I'm normally happy to respond. :smile:

**Comment by m-toman** [comment]
Hi, how are things going?

I am also on this and I modified the upsampling layers as follows:
``` 
    "upsample_rates": [8,8,4,2],                                                                                                        
    "upsample_kernel_sizes": [16,16,8,4],    
```
that worked quite well with MelGAN for me.
As I am also using a different preprocessing and few other things differently I am currently testing with LJ only before going on to a universal model.

Someone tried the v2 config?

**Comment by CookiePPP** [comment]
> Hi, how are things going?

Currently waiting for my tacotron to retrain with new speakers. One problem I found is that my tacotron2 doesn't provide good enough spectrograms, so when I fine-tune HiFi-GAN on a 350~ speaker tacotron2, the output audio seems slightly blended between speakers. (it's subtle, but noticeable when you listen to audio samples side-by-side)
I've added a GAN loss to my tacotron2 to significantly reduce the blur and changed the `n_mel_channels` to 160 and everything is currently training again. 😄

**Comment by CookiePPP** [comment]
> Someone tried the v2 config?

I don't know, but I'm somewhat curious. I lose half my processing time to the vocoder so if v2 sounds good at 44Khz then it's definitely worth keeping a pretrained copy lying around.

**Comment by m-toman** [comment]
I am now training with v2. On CPU I then get the same performance as MelGAN. So for some test sentence this is from 1.3s to 0.3s.
Only have 1 gpu free atm so it might take ages unfortunately.

**Comment by m-toman** [comment]
Update, my curve is definitely flatter, hope it will continue to decrease. The two times where it restarted was when I switched from v1 to v2 and another reset where I had a problem writing the checkpoint.

![image](https://user-images.githubusercontent.com/9433378/100454098-86e3e800-30bc-11eb-98af-7028d0d35084.png)

**Comment by m-toman** [comment]
Quick update again. I am at more than 400k steps and still sounds rather noisy. Loss still decreasing so let's see after the weekend.
It's V2 at 44kHz so either takes longer  or the model capacity is not enough.

**Comment by charlie-codes-choi** [comment]
Curious how it went @m-toman !

**Comment by m-toman** [comment]
> Curious how it went @m-toman !

Had to stop a few times because of more urgent trainings and recently restarted with additional discriminator periods. Get okish quality pretty quickly but it can't seem to get rid of quite a few artefacts.

**Comment by v-nhandt21** [comment]
I experiment with segment = 16384, while using the sample rate 22050, hope that work and still waitin g for the réults

**Comment by skol101** [comment]
@CookiePPP how did you changed VCTK dataset (all wavs and texts) to the acceptable form by this code?

**Comment by CookiePPP** [comment]
@skol101 
I changed the code.
